crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -134,6 +230,116 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for BluesnapPaymentsRequest {
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "apple_pay_metadata",
+            })?
+            .expose();
+        let applepay_metadata = apple_pay_metadata
+            .parse_value::<api_models::payments::ApplepaySessionTokenData>(
+                "ApplepaySessionTokenData",
+            )
+            .change_context(errors::ConnectorError::ParsingFailed)?;
+        Ok(Self {
+            wallet_type: "APPLE_PAY".to_string(),
+            validation_url: consts::APPLEPAY_VALIDATION_URL.to_string().into(),
+            domain_name: applepay_metadata.data.session_token_data.initiative_context,
+            display_name: Some(applepay_metadata.data.session_token_data.display_name),
+        })
+    }
+}
+
+impl TryFrom<types::PaymentsSessionResponseRouterData<BluesnapWalletTokenResponse>>
+    for types::PaymentsSessionRouterData
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::PaymentsSessionResponseRouterData<BluesnapWalletTokenResponse>,
+    ) -> Result<Self, Self::Error> {
+        let response = &item.response;
+
+        let wallet_token = consts::BASE64_ENGINE
+            .decode(response.wallet_token.clone().expose())
+            .into_report()
+            .change_context(errors::ConnectorError::ParsingFailed)?;
+
+        let session_response: api_models::payments::ApplePaySessionResponse = wallet_token[..]
+            .parse_struct("ApplePayResponse")
+            .change_context(errors::ConnectorError::ParsingFailed)?;
+
+        let metadata = item
+            .data
+            .connector_meta_data
+            .clone()
+            .get_required_value("metadata")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "metadata",
+            })?
+            .expose();
+        let applepay_metadata = metadata
+            .parse_value::<api_models::payments::ApplepaySessionTokenData>(
+                "ApplepaySessionTokenData",
+            )
+            .change_context(errors::ConnectorError::ParsingFailed)?;
+
+        Ok(Self {
+            response: Ok(types::PaymentsResponseData::SessionResponse {
+                session_token: types::api::SessionToken::ApplePay(Box::new(
+                    api_models::payments::ApplepaySessionTokenResponse {
+                        session_token_data: session_response,
+                        payment_request_data: api_models::payments::ApplePayPaymentRequest {
+                            country_code: item
+                                .data
+                                .request
+                                .country
+                                .get_required_value("country_code")
+                                .change_context(errors::ConnectorError::MissingRequiredField {
+                                    field_name: "country_code",
+                                })?,
```

Comment:
```suggestion
                            country_code: item.data.request.get_country()?,
```

Reply:
`get_country` not implemented for struct `PaymentsSessionData